evap/staff/importers.py

```diff
@@ -402,18 +402,25 @@ def process_evaluation(self, evaluation_data, sheetname, row):
                 self.evaluations[evaluation_id] = evaluation_data
                 self.names_de.add(evaluation_data.name_de)
         else:
-            if evaluation_data.equals_except_for_degrees(self.evaluations[evaluation_id]):
+            known_data = self.evaluations[evaluation_id]
+            if evaluation_data.equals_except_for_degrees(known_data):
                 self.warnings[ImporterWarning.DEGREE].append(
                     _(
                         'Sheet "{}", row {}: The course\'s "{}" degree differs from it\'s degree in a previous row.'
                         " Both degrees have been set for the course."
                     ).format(sheetname, row + 1, evaluation_data.name_en)
                 )
-                self.evaluations[evaluation_id].degrees |= evaluation_data.degrees
-                self.evaluations[evaluation_id].errors = merge_dictionaries_of_sets(
-                    self.evaluations[evaluation_id].errors, evaluation_data.errors
-                )
-            elif evaluation_data != self.evaluations[evaluation_id]:
+
+                known_data.degrees |= evaluation_data.degrees
+
+                assert all(key in ["degrees", "course_type", "is_graded"] for key in evaluation_data.errors)
```

Comment:
you could use set semantics, if you wanted
```python3
assert evaluation_data.errors.keys() <= {"degrees", "course_type", "is_graded"}
```

Reply:
nice